### DISCUSSION TITLE
log file not getting generated after migrating to java 17

### DISCUSSION THREAD

**Discussion by ravadisagarrao** [qos-ch/logback#848] [General]
After upgrading our application from java 8 to Java 17 log file is not getting generated with logback version 1.3.14

When searched in google found that 1.3.14 supports till Java 8, so added latest logback version 1.5.7 in pom.xml but still log file is not getting generated.

Could someone please review below logback.xml config file and let me know whether am I missing something?

<?xml version="1.0" encoding="UTF-8"?>
<configuration debug="false">

    <contextName>cmd-cxp-api</contextName>
    <jmxConfigurator />
<property file="${config.file.location}" />
    <appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
        
        <encoder>
            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %X{tenant.id} %logger{36} - %msg%n</pattern>
        </encoder>
    </appender>

    <appender name="FILE" class="ch.qos.logback.core.rolling.RollingFileAppender">
        <file>${log.file.location}/emcd_server.log</file>
        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
            
            <fileNamePattern>${log.file.location}/emcd_server.%d{yyyy-MM-dd}.%i.log</fileNamePattern>
            <timeBasedFileNamingAndTriggeringPolicy class="ch.qos.logback.core.rolling.SizeAndTimeBasedFNATP">
                
                <maxFileSize>10MB</maxFileSize>
            </timeBasedFileNamingAndTriggeringPolicy>
            
            <maxHistory>30</maxHistory>
        </rollingPolicy>
        <encoder>
            <pattern>%date %level %X{tenant.id} [%thread] %logger{10} [%file:%line] %msg%n</pattern>
        </encoder>
    </appender>

    <logger name="com.xxx.mobile">
        <level value="${log.level}" />
    </logger>

    <logger name="org.springframework">
        <level value="WARN" />
    </logger>

    <logger name="org.apache">
        <level value="WARN" />
    </logger>

    <root level="WARN">
        <appender-ref ref="STDOUT" />
        <appender-ref ref="FILE" />
    </root>

</configuration>

Thanks,
Sagar

**Comment by i-make-robots** [comment]
Hello, everyone. I hope you're doing well.

I believe I have a similar issue.

My Java 22 open source project is https://github.com/MarginallyClever/Makelangelo-software/.
I'm using logback and slf4j with a rolling file appender. My config file is https://github.com/MarginallyClever/Makelangelo-software/blob/master/src/main/resources/logback.xml
I then use JPackage to distribute my app with all the dependencies baked into a fat jar.

For MOST people the log file appears and fills normally. For a select number of users the log file appears but stays empty.
They have been gracious enough to try running the JAR from the command line but we've seen no extra console messages.
Running in administrator mode doesn't help.
Immediate flushing doesn't help.
Adding a logback status listener (logback-status.log) also doesn't appear for those tricky users.

Can someone please give me a hint what I'm doing wrong?
